src/foam/dao/PublicUserInfoDAO.java

```diff
@@ -0,0 +1,112 @@
+package foam.dao;
+
+import foam.dao.*;
+import foam.core.FObject;
+import foam.core.PropertyInfo;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.dao.Sink;
+import foam.nanos.auth.PublicUserEntity;
+import foam.nanos.auth.User;
+import net.nanopay.tx.model.Transaction;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.logger.Logger;
+
+/**
+ * Populate a model returned from find or select with a small subset of public 
+ * user information.
+ *
+ * Example: A call like
+ *
+ *   new foam.dao.PublicUserInfoDAO(x, "payerId", "payer", ...)
+ *
+ * will look for a property named "payerId" on the object being selected and
+ * look up the user associated with that id. It will then set the given property
+ * ("payer") to the object being returned that contains some public properties
+ * of the associated user.
+ *
+ * Requires both properties to already be defined on the model. For example, on
+ * Transaction.js:
+ *
+ *     ...
+ *   },
+ *   {
+ *    class: 'FObjectProperty',
+ *    of: 'net.nanopay.tx.model.TransactionEntity',
+ *    name: 'payee',
+ *    storageTransient: true
+ *   },
+ *   {
+ *    class: 'Long',
+ *    name: 'payeeId'
+ *   },
+ *   {
+ *     ...
+ */
+public class PublicUserInfoDAO
+  extends ProxyDAO
+{
+  private String idPropertyName_;
+  private String propertyNameToSet_;
+
+  public PublicUserInfoDAO(
+      X x,
+      String idPropertyName,
```

Comment:
Rather than passing in property names which won't be checked by the compiler, you can pass in the PropertyInfo's directly. Ex.: User.ID, or Transaction.PAYEE_INFO. Make the type be foam.mlang.F, which propertyInfo's implement then you can call idProperty_.f(clone) instead of clone.getProperty(idPropertyName_), with the first option being both shorter and faster.

Reply:
That was my first thought, but I didn't know if it could be done. Right now it would look like this in the services file:

`new foam.dao.PublicUserInfoDAO(x, "payerId", "payer", ...)`

Would the new one look like the following? Will those properties be defined by the time that code is executed?

`new foam.dao.PublicUserInfoDAO(x, net.nanopay.tx.model.PAYER_ID, net.nanopay.tx.model.PAYER, ...)`

My other question was how to write to the second property. I know you can read a value with the `.f()` method, but how would I write to net.nanopay.tx.model.PAYER, for example?